### DISCUSSION TITLE
How to implement a custom User table?

### DISCUSSION THREAD

**Discussion by yurix73** [supabase/supabase#16742] [Questions]
I've played around with Supabase for some time now and I'd like to add / change some columns of the default User table, but it seems like there is no way to do this via the CLI, is this correct? Stuff like adding a "roles" attribute or changing the eMail attribute to a username.

Is this the intended way of using supabase? I tried creating another User table but then I guess I need to restrict who can make GET-Requests to that table and handle all of the password hashing in Client-Side JS. Is that secure?

Thanks for your help :)

**Comment by GaryAustin1** [comment]
You should not modify anything in the auth schema tables.
The user management starters in the guides and the Welcome tab of the SQL editor show using a trigger to make a secondary user table.  You can protect that table as desired and use it in joins or RLS if desired.
